app/controllers/lti_launch_controller.rb

```diff
@@ -0,0 +1,224 @@
+class LtiLaunchController < ApplicationController
+  respond_to :json
+  skip_before_action :set_course
+  skip_before_action :authorize_user_for_course
+  skip_before_action :update_persistent_announcements
+  skip_before_action :authenticate_for_action
+
+  # have to do because we are making a POST request from Canvas
+  skip_before_action :verify_authenticity_token
+
+  # action_auth_level :launch, :instructor
+  class LtiError < StandardError
+    def initialize(msg, status_code = :bad_request)
+      @status_code = status_code
+      super(msg)
+    end
+  end
+  rescue_from LtiError, with: :respond_with_lti_error
+  respond_to :json
```

Comment:
Is this line redundant given the `respond_to` on line 2?

Reply:
Yep, good catch